### ISSUE TITLE
After you dismiss "rename mode" for a symbol that is not a type definition, you never get asked to rename it again

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#18622]
When you change the name of a symbol, Roslyn offers to rename all references to that symbol to another symbol. This only works the very first time however, once you've dismissed it - you never get offered again to rename.

``` C#
using System.Collections;


public interface IProjectConfigurationsService
{
}

class Program : IProjectConfigurationsService
{

}
```

1. Highlight `IProjectConfigurationsService` in `Program : IProjectConfigurationsService`
2. Type `ICollection`

-- Notice the rename frame is around ICollection --
![image](https://cloud.githubusercontent.com/assets/1103906/24942651/b719a078-1f95-11e7-9ba9-9dbb6c9ff3b6.png)

3. Press ESC to dimiss completion window and then ESC to get out of rename mode
4. CTRL+Z twice to undo the name and change it back to 'IProjectConfigurationsService'
5. CTRL+Y to redo the name

Expected: To be back in "rename mode"
Actual: Not in rename mode,

NOTE: If you do this to type definition itself `interface IProjectConfigurationsService` then this works correctly. Note, while above is a good repro, it doesn't really matter how you rename the symbol in the future - you never get into rename mode again for that symbol unless I close and reopen the file.

**Comment by CyrusNajmabadi** [comment]
> you never get into rename mode again for that symbol unless I close and reopen the file.

Does this include going and making another edit in the same file, and then going back and editing this symbol name?  Or does it only truly work again if the file is closed/opened.

**Comment by davkean** [comment]
You are never offered again as far as I can tell, even when I cut the type and paste into _another_ file.

You don't even need undo or ESC to be involved - I found I could repro it simply by making an unrelated change after step 2.

**Comment by davkean** [comment]
Actually it's worse - it looks like state is broken for the entire project, I can't enter rename mode again if I have any file open.

**Comment by CyrusNajmabadi** [comment]
Sounds like something broken in the state machine tracking.  First thing to fix will be making it so that you're not in a borked state where rename simply doesn't work again.  Then, it would be a nice-to-have to update state tracking to be better at understanding undo/redo so you get rename offered in cases like this.

**Comment by CyrusNajmabadi** [comment]
Tagging @jasonmalinowski in case he has any ideas what would have happened with the rename tracking state machine.

**Comment by Pilchie** [comment]
@dpoeschl Wrote the rename tracking state machine.

**Comment by dpoeschl** [comment]
@davkean I can't get this to repro. Is it still happening for you? Can you make a gif or something?

**Comment by davkean** [comment]
Yeah can still repro this:

![rename](https://cloud.githubusercontent.com/assets/1103906/25605938/c4bffcf6-2f51-11e7-9b94-b18cc566d8b2.gif)

**Comment by dpoeschl** [comment]
That is the lowest resolution gif I have ever seen. Is bandwidth down under *that* bad?

**Comment by davkean** [comment]
GitHub has a 10 MB limit on files.

**Comment by Pilchie** [comment]
I can't repro this either.   Very strange.

**Comment by davkean** [comment]
This is still a 100% repro for me, are you sure you are following the repro steps exactly?

Here is it again to show that I'm not insane:

![rename](https://cloud.githubusercontent.com/assets/1103906/26331043/3e74d440-3f92-11e7-823b-8391db9a15c3.gif)

**Comment by Pilchie** [comment]
@davkean in step 5 I could only undo once, but it just worked with a single undo there.

**Comment by davkean** [comment]
Yeah, I just repro'd with a single undo and it still repro'd.

**Comment by davkean** [comment]
This is still a 100% repro for me.

**Comment by davkean** [comment]
Still running into renaming tracking issues - basically gets into a state that stops me from it ever recognizing that I've renamed a symbol.

**Comment by CyrusNajmabadi** [comment]
I also can't repro this on latest d18 builds.
